ament_cmake_clang_format/cmake/ament_cmake_clang_format_lint_hook.cmake

```diff
@@ -23,6 +23,10 @@ file(GLOB_RECURSE _source_files FOLLOW_SYMLINKS
   "*.hxx"
 )
 if(_source_files)
+  if(DEFINED ament_cmake_clang_format_CONFIG_FILE)
+    set(config_file_path ${ament_cmake_clang_format_CONFIG_FILE})
+  endif()
```

Comment:
In the current form the global variable is only considered by this auto-lint hook. The function doc suggests that it also affect any manual call of `ament_clang_format`.

Depending on which behavior is desired either the docs have to be changed or code has to be modified to move the usage of the global variable into the `ament_clang_format` function. I am leaning slightly toward the second one - since it is easier to document.

`cppcheck` might need the same fix since it looks like the pattern there is the same.

Reply:
If there is a case in which both the `CONFIG_FILE` and `ament_cmake_clang_format_CONFIG_FILE` are defined, which of the two should be used?  I'm leaning towards `CONFIG_FILE` given that's the most explicit config path specified by the user at the time of calling

Reply:
> If there is a case in which both `CONFIG_FILE` and `ament_cmake_clang_format_CONFIG_FILE` are defined

Could be.

> I'm leaning towards CONFIG_FILE given that's the most explicit config path specified by the user at the time of calling

:+1: